sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java

```diff
@@ -102,73 +102,82 @@ public DisruptorEvent newInstance() {
     this.blocking = blocking;
   }
 
-  void enqueue(ReadableSpan readableSpan, EventType eventType) {
-    enqueue(readableSpan, eventType, null);
-  }
-
-  // Enqueues an event on the {@link DisruptorEventQueue}.
-  void enqueue(ReadableSpan readableSpan, EventType eventType, CountDownLatch flushLatch) {
+  void enqueueStartEvent(ReadableSpan span) {
     if (isShutdown) {
       if (!loggedShutdownMessage.getAndSet(true)) {
         logger.info("Attempted to enqueue entry after Disruptor shutdown.");
       }
       return;
     }
+    enqueue(EventType.ON_START, span, null);
+  }
 
-    if (blocking) {
-      ringBuffer.publishEvent(TRANSLATOR_THREE_ARG, eventType, readableSpan, flushLatch);
-    } else {
-      // TODO: Record metrics if element not added.
-      ringBuffer.tryPublishEvent(TRANSLATOR_THREE_ARG, eventType, readableSpan, flushLatch);
+  void enqueueEndEvent(ReadableSpan span) {
+    if (isShutdown) {
+      if (!loggedShutdownMessage.getAndSet(true)) {
+        logger.info("Attempted to enqueue entry after Disruptor shutdown.");
+      }
```

Comment:
this is duplicated code. can it be moved to a method?

Reply:
Customized the messages to be more relevant.